### PR TITLE
Default token store implementation

### PR DESCRIPTION
**Pull request by KarthikShanmugasundaram**

### CODE REVIEW

File: lib/box-node-sdk.js [hunk 1]
```diff
@@ -53,6 +53,7 @@ var EventEmitter = require('events').EventEmitter,
 	util = require('util'),
 	qs = require('querystring'),
 	Config = require('./util/config'),
+	TokenStore = require('./util/token-store'),
```
Comment (mattwiller) [inline lib/box-node-sdk.js]: Consider naming this variable `MemoryTokenStore` for clarity

File: lib/box-node-sdk.js [hunk 2]
```diff
@@ -176,6 +177,16 @@ BoxSDKNode.prototype.getAppAuthClient = function(type, id) {
...
 
 /**
+ * Create the in-memory token store instance for the user
+ *
+ * @param {string} userId the ID of the user whose token info can be persisted in-memory.
+ * @returns {TokenStore} A new TokenStore instance
+ */
+BoxSDKNode.prototype.getTokenStore = function(userId) {
+	return new TokenStore(userId);
+};
```
Comment (mattwiller) [inline lib/box-node-sdk.js]: I would simply provide a reference to the class here, so the client can do whatever they want with it (not everyone will want to use userID necessarily.

```js
BoxSDKNode.MemoryTokenStore = MemoryTokenStore;
```

File: lib/util/token-store.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+ * @module box-node-sdk/token-store
+ */
+
+'use strict';
+
+var store = new Map();
+
+/**
+ * Basic in-memory Token Store, not suitable for use in production!
+ * @param {string} userID The ID of the user whose tokens will be stored
```
Comment (mattwiller) [inline lib/util/token-store.js]: Consider renaming; user ID is not necessary, any identifier will do.  For example, some applications use an internal session ID instead!

File: tests/integration-test.js [hunk 4]
```diff
@@ -10,14 +10,13 @@
...
 	http = require('http'),
-	leche = require('leche'),
 	sinon = require('sinon'),
 	nock = require('nock'),
 	fs = require('fs'),
 	path = require('path'),
 	jwt = require('jsonwebtoken');
 
-describe('Box Node SDK', function() {
+describe.only('Box Node SDK', function() {
```
Comment (mattwiller) [inline tests/integration-test.js]: Need to remove the `.only` before merging

File: tests/integration-test.js [hunk 5]
```diff
@@ -173,9 +172,14 @@ describe('Box Node SDK', function() {
...
 
+		var sdk = new BoxSDK({
+			clientID: TEST_CLIENT_ID,
+			clientSecret: TEST_CLIENT_SECRET
+		});
+
 		var fileID = '98740596456',
 			fileName = 'Test Document.pdf',
-			tokenStoreFake = leche.create(['read', 'write', 'clear']),
+			tokenStoreFake = sdk.getTokenStore(),
```
Comment (mattwiller) [inline tests/integration-test.js]: This doesn't need to be changed

File: lib/util/token-store.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+ * @constructor
+ */
+function TokenStore(userID) {
```
Comment (mattwiller) [inline lib/util/token-store.js]: Consider renaming this to `MemoryTokenStore` as well for consistency/clarity

File: lib/box-node-sdk.js (hunk 1 above)
Comment (KarthikShanmugasundaram) [inline lib/box-node-sdk.js]: done

File: lib/box-node-sdk.js (hunk 2 above)
Comment (KarthikShanmugasundaram) [inline lib/box-node-sdk.js]: done

File: lib/util/token-store.js (hunk 3 above)
Comment (KarthikShanmugasundaram) [inline lib/util/token-store.js]: done

File: lib/util/token-store.js (hunk 6 above)
Comment (KarthikShanmugasundaram) [inline lib/util/token-store.js]: done

File: tests/integration-test.js (hunk 4 above)
Comment (KarthikShanmugasundaram) [inline tests/integration-test.js]: done

File: lib/box-node-sdk.js [hunk 7]
```diff
@@ -118,6 +119,13 @@ BoxSDKNode.collaborationRoles = BoxSDKNode.prototype.collaborationRoles = BoxCli
 BoxSDKNode.CURRENT_USER_ID = BoxSDKNode.prototype.CURRENT_USER_ID = BoxClient.prototype.CURRENT_USER_ID;
 
 /**
+ * MemoryTokenStore is a default implementation of persistent token store
+ * @static
+ * @type {Object}
```
Comment (mattwiller) [inline lib/box-node-sdk.js]: This is the wrong type; MemoryTokenStore is a constructor function

File: lib/util/memory-token-store.js [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+'use strict';
+
+var store = new Map();
+
+/**
+ * Basic in-memory Token Store, not suitable for use in production!
+ * @param {string} ID a unique identifier key for which the token is the value.
+ * @constructor
+ */
+function MemoryTokenStore(ID) {
```
Comment (mattwiller) [inline lib/util/memory-token-store.js]: Variables should be camel-cased, e.g. `id`

**Pull request closed without merging**